diff -ur gnome-control-center-3.36.0.orig/meson.build gnome-control-center-3.36.0/meson.build
--- gnome-control-center-3.36.0.orig/meson.build	2020-03-07 02:17:04.000000000 +0100
+++ gnome-control-center-3.36.0/meson.build	2020-03-11 18:02:10.086852496 +0100
@@ -205,7 +205,8 @@
 config_h.set('HAVE_SNAP', enable_snap,
              description: 'Define to 1 if Snap support is enabled')
 
-if host_is_linux
+enable_networkmanager = get_option('networkmanager')
+if enable_networkmanager
   # network manager
   network_manager_deps = [
     dependency('libnm', version: '>= 1.12.0'),
@@ -213,33 +214,39 @@
     dependency('mm-glib', version: '>= 0.7')
   ]
 endif
-config_h.set('BUILD_NETWORK', host_is_linux,
+config_h.set('BUILD_NETWORK', enable_networkmanager,
              description: 'Define to 1 to build the Network panel')
-config_h.set('HAVE_NETWORK_MANAGER', host_is_linux,
+config_h.set('HAVE_NETWORK_MANAGER', enable_networkmanager,
              description: 'Define to 1 if NetworkManager is available')
 
+enable_bluetooth = get_option('bluetooth')
+enable_wacom = get_option('wacom')
 if host_is_linux_not_s390
   # gnome-bluetooth
-  gnome_bluetooth_dep = dependency('gnome-bluetooth-1.0', version: '>= 3.18.2')
-
-  libwacom_dep = dependency('libwacom', version: '>= 0.7')
-
-  wacom_deps = [
-    libwacom_dep,
-  ]
-  config_h.set('HAVE_WACOM_3D_STYLUS', libwacom_dep.version().version_compare('>= 0.27'),
+  if enable_bluetooth
+    gnome_bluetooth_dep = dependency('gnome-bluetooth-1.0', version: '>= 3.18.2')
+  endif
+
+  if enable_wacom
+    libwacom_dep = dependency('libwacom', version: '>= 0.7')
+
+    wacom_deps = [
+      libwacom_dep,
+    ]
+    config_h.set('HAVE_WACOM_3D_STYLUS', libwacom_dep.version().version_compare('>= 0.27'),
                description: 'Define to 1 if libwacom provides definition for 3D styli')
+  endif
 else
   message('Bluetooth and Wacom panels will not be built (no USB support on this platform)')
   message('Thunderbolt panel will not be built (not supported on this platform)')
 endif
-config_h.set('BUILD_BLUETOOTH', host_is_linux_not_s390,
+config_h.set('BUILD_BLUETOOTH', enable_bluetooth,
              description: 'Define to 1 to build the Bluetooth panel')
-config_h.set('HAVE_BLUETOOTH', host_is_linux_not_s390,
+config_h.set('HAVE_BLUETOOTH', enable_bluetooth,
              description: 'Define to 1 if bluetooth support is available')
-config_h.set('BUILD_WACOM', host_is_linux_not_s390,
+config_h.set('BUILD_WACOM', enable_wacom,
              description: 'Define to 1 to build the Wacom panel')
-config_h.set('HAVE_WACOM', host_is_linux_not_s390,
+config_h.set('HAVE_WACOM', enable_wacom,
              description: 'Define to 1 if Wacom is supportted')
 config_h.set('BUILD_THUNDERBOLT', host_is_linux_not_s390,
              description: 'Define to 1 to build the Thunderbolt panel')
@@ -291,11 +298,11 @@
 output += '     Tracing .................................... ' + enable_tracing.to_string() + '\n'
 output += '     Optimized .................................. ' + control_center_optimized.to_string() + '\n'
 output += ' Panels \n'
-output += '     GNOME Bluetooth (Bluetooth panel) .......... ' + host_is_linux_not_s390.to_string() + '\n'
+output += '     GNOME Bluetooth (Bluetooth panel) .......... ' + enable_bluetooth.to_string() + '\n'
 output += '     Cheese (Users panel webcam support) ........ ' + enable_cheese.to_string() + '\n'
 output += '     IBus (Region panel IBus support) ........... ' + enable_ibus.to_string() + '\n'
-output += '     NetworkManager (Network panel) ............. ' + host_is_linux.to_string() + '\n'
-output += '     Wacom (Wacom tablet panel) ................. ' + host_is_linux_not_s390.to_string() + '\n'
+output += '     NetworkManager (Network panel) ............. ' + enable_networkmanager.to_string() + '\n'
+output += '     Wacom (Wacom tablet panel) ................. ' + enable_wacom.to_string() + '\n'
 output += '     Snap support ............................... ' + enable_snap.to_string() + '\n'
 
 message(output)
diff -ur gnome-control-center-3.36.0.orig/meson_options.txt gnome-control-center-3.36.0/meson_options.txt
--- gnome-control-center-3.36.0.orig/meson_options.txt	2020-03-07 02:17:04.000000000 +0100
+++ gnome-control-center-3.36.0/meson_options.txt	2020-03-11 17:59:53.478352423 +0100
@@ -7,3 +7,8 @@
 option('tracing', type: 'boolean', value: false, description: 'add extra debugging information')
 option('wayland', type: 'boolean', value: true, description: 'build with Wayland support')
 option('profile', type: 'combo', choices: ['default','development'], value: 'default')
+option('networkmanager', type: 'boolean', value: true, description: 'build with networkmanager support')
+option('wacom', type: 'boolean', value: true, description: 'build with wacom support')
+option('bluetooth', type: 'boolean', value: true, description: 'build with bluetooth support')
+option('kerberos', type: 'boolean', value: true, description: 'build with kerberos support')
+
diff -ur gnome-control-center-3.36.0.orig/panels/meson.build gnome-control-center-3.36.0/panels/meson.build
--- gnome-control-center-3.36.0.orig/panels/meson.build	2020-03-07 02:17:04.000000000 +0100
+++ gnome-control-center-3.36.0/panels/meson.build	2020-03-11 17:56:52.449007374 +0100
@@ -29,15 +29,21 @@
   'user-accounts'
 ]
 
-if host_is_linux
+if enable_networkmanager
   panels += ['network']
 endif
 
+if enable_bluetooth
+  panels += ['bluetooth']
+endif
+
+if enable_wacom
+  panels += ['wacom']
+endif
+
 if host_is_linux_not_s390
   panels += [
-    'bluetooth',
     'thunderbolt',
-    'wacom'
   ]
 endif
 
Nur in gnome-control-center-3.36.0/panels: meson.build.orig.
diff -ur gnome-control-center-3.36.0.orig/panels/user-accounts/cc-realm-manager.c gnome-control-center-3.36.0/panels/user-accounts/cc-realm-manager.c
--- gnome-control-center-3.36.0.orig/panels/user-accounts/cc-realm-manager.c	2020-03-07 02:17:04.000000000 +0100
+++ gnome-control-center-3.36.0/panels/user-accounts/cc-realm-manager.c	2020-03-11 17:56:52.450007370 +0100
@@ -22,7 +22,9 @@
 
 #include "cc-realm-manager.h"
 
+#ifdef HAVE_KERBEROS
 #include <krb5/krb5.h>
+#endif
 
 #include <glib.h>
 #include <glib/gi18n.h>
@@ -613,6 +615,7 @@
         g_slice_free (LoginClosure, login);
 }
 
+#ifdef HAVE_KERBEROS
 static krb5_error_code
 login_perform_kinit (krb5_context k5,
                      const gchar *realm,
@@ -676,6 +679,7 @@
 
         return code;
 }
+#endif /* HAVE_KERBEROS */
 
 static void
 kinit_thread_func (GTask *task,
@@ -683,6 +687,7 @@
                    gpointer task_data,
                    GCancellable *cancellable)
 {
+#ifdef HAVE_KERBEROS
         LoginClosure *login = task_data;
         krb5_context k5 = NULL;
         krb5_error_code code;
@@ -761,7 +766,10 @@
 
         if (k5)
                 krb5_free_context (k5);
-
+#else /* HAVE_KERBEROS */
+        g_task_return_new_error (task, CC_REALM_ERROR, CC_REALM_ERROR_GENERIC,
+                                         _("gnome-control-center was built without kerberos support"));
+#endif
         g_object_unref (task);
 }
 
diff -ur gnome-control-center-3.36.0.orig/panels/user-accounts/meson.build gnome-control-center-3.36.0/panels/user-accounts/meson.build
--- gnome-control-center-3.36.0.orig/panels/user-accounts/meson.build	2020-03-07 02:17:04.000000000 +0100
+++ gnome-control-center-3.36.0/panels/user-accounts/meson.build	2020-03-11 17:56:52.450007370 +0100
@@ -156,20 +156,24 @@
   'um-fingerprint-dialog.c',
 )
 
-# Kerberos support
-krb_dep = dependency('krb5', required: false)
-assert(krb_dep.found(), 'kerberos libraries not found in your path')
-
 deps = common_deps + [
   accounts_dep,
   gdk_pixbuf_dep,
   gnome_desktop_dep,
   liblanguage_dep,
-  krb_dep,
   m_dep,
   polkit_gobject_dep,
   dependency('pwquality', version: '>= 1.2.2')
 ]
+# Kerberos support
+enable_kerberos = get_option('kerberos')
+if enable_kerberos
+  krb_dep = dependency('krb5', required: false)
+  assert(krb_dep.found(), 'kerberos libraries not found in your path')
+  deps += krb_dep
+  config_h.set('HAVE_kerberos', enable_kerberos,
+             description: 'Define to 1 to enable kerberos support')
+endif
 
 if enable_cheese
   deps += cheese_deps
diff -ur gnome-control-center-3.36.0.orig/shell/meson.build gnome-control-center-3.36.0/shell/meson.build
--- gnome-control-center-3.36.0.orig/shell/meson.build	2020-03-07 02:17:05.000000000 +0100
+++ gnome-control-center-3.36.0/shell/meson.build	2020-03-11 17:56:52.451007367 +0100
@@ -119,7 +119,7 @@
   shell_deps += cheese_deps
 endif
 
-if host_is_linux_not_s390
+if enable_wacom
   shell_deps += wacom_deps
 endif
 
Nur in gnome-control-center-3.36.0/shell: meson.build.orig.
